topics/08_ai_integration_basic/main_test_automation_integrating_ai.md

```diff
@@ -1,3 +1,100 @@
+2. [GPT-3](https://openai.com/research/gpt-3/): GPT-3 (**G**enerative **P**re-trained **T**ransformer 3) is a language model capable of generating human-like text based on prompts or questions. It has been used for various tasks such as writing articles, answering queries, and even generating code snippets. Nowadays, there are several models with the same purposes. Below is a comparison of popular generative AI models and their use cases:
+
+    | **Model**                  | **Main Differences**                                                                                 | **When to Use**                                                              | **When to Avoid**                                                          |
+    |:---------------------------|:-----------------------------------------------------------------------------------------------------|:-----------------------------------------------------------------------------|:---------------------------------------------------------------------------|
+    | **GPT-3 (OpenAI)**         | Language model for generating human-like text based on prompts or questions.                         | Use when generating text for various tasks such as writing or querying.      | Avoid when specific domain expertise or fine-grained control is needed.    |
+    | **Gemini (Google)**        | More focused on code generation and assistance for software development tasks.                       | Suitable for generating code snippets and assisting in programming tasks.    | Not ideal for non-programming tasks or general text generation.            |
+    | **Bing Model (Microsoft)** | Developed for search and text-based applications, providing AI-generated suggestions and responses.  | Well-suited for enhancing search experiences or generating recommendations.  | May not be suitable for creative content generation or specialized tasks.  |
+
+3. [DALL-E](https://openai.com/research/dall-e/): Developed by OpenAI, DALL-E can generate original images from textual descriptions. It can create unique and imaginative visuals from text prompts that were never seen during training.
+
+4. [DeepDream](https://deepdreamgenerator.com/): DeepDream produces fascinating and surreal images by applying neural network patterns back onto an input image. It can create dream-like visuals by magnifying and enhancing specific features of the original image.
+
+5. [MuseNet](https://openai.com/blog/musenet/): MuseNet is an AI model that generates original musical compositions in various styles and genres. It can create harmonies, melodies, and even orchestral arrangements that are indistinguishable from human-composed pieces.
+
+
+### GitHub Copilot
+
+GitHub Copilot is an AI-powered code completion tool developed by GitHub in collaboration with OpenAI. It uses a generative AI model to assist developers by automatically suggesting code completions and generating code snippets based on the context within their code editors.
```

Comment:
If you are going to do a live demo on how to use GitHub Copilot, it would be good to attach code examples with prompts, for example hot to create a method or how to refactor a method, so the mentees could play with them

Reply:
I will think something to add. Maybe we could add just a few things here and merge, them we can work on it to improve after the training because for this one, no time anymore. :(

Reply:
@ericrommel sure, you can merge it, but [IMHO] I would mark this topic as '[WIP]' as it provides just some references and very high-level overview. If there is plan to use this book in the future, the topic needs to be extended.